common/environment/environment.cpp

```diff
@@ -1103,6 +1159,57 @@ class CConstDropZoneInfo : public CConstEnvBase, implements IConstDropZoneInfo
     StringBuffer posixPath;
 };
 
+class CConstSparkThorInfo : public CConstEnvBase, implements IConstSparkThorInfo
+{
+public:
+    IMPLEMENT_IINTERFACE;
+    IMPLEMENT_ICONSTENVBASE;
+    CConstSparkThorInfo(CLocalEnvironment *env, IPropertyTree *root) : CConstEnvBase(env, root) {}
+
+    virtual IStringVal &getBuild(IStringVal &str) const
+    {
+        str.set(root->queryProp("@build"));
```

Comment:
is it possible root doesn't contain any of these attributes? if so is it worthwhile to verify if hasprop before setting the str?

Reply:
The set() can handle a nullptr. I do not need to verify here.